src/harness/testcases/WINNF_FT_S_DRG_testcase.py

```diff
@@ -0,0 +1,127 @@
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+from datetime import datetime
+import json
+import os
+import unittest
+
+import sas
+from util import winnforum_testcase
+
+class DeregistrationTestcase(unittest.TestCase):
+
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_DRG_1(self):
+    """Valid and correct CBSD ID: two deregistrationRequest objects
+    CBSD sends deregistration request to SAS with two deregistrationRequest
+    objects that contain correct and valid CBSD ID, the response should be
+    SUCCESS.
+    """
+
+    # Register the devices
+    registration_request = []
+    for device_filename in ('device_a.json', 'device_c.json'):
+      device = json.load(
+        open(os.path.join('testcases', 'testdata', device_filename)))
+      self._sas_admin.InjectFccId({'fccId': device['fccId']})
+      self._sas_admin.InjectUserId({'userId': device['userId']})
+      registration_request.append(device)
+    request = {'registrationRequest': registration_request}
+    response = self._sas.Registration(request)['registrationResponse']
+
+    # Check registration response
+    cbsd_ids = []
+    self.assertEqual(len(response), 2)
+    for resp in response:
+      self.assertEqual(resp['response']['responseCode'], 0)
+      cbsd_ids.append(resp['cbsdId'])
+    del request, response
+
+    # Deregister the device
+    request = {'deregistrationRequest': [{'cbsdId': cbsd_ids[0]}, {'cbsdId': cbsd_ids[1]}]}
+    response = self._sas.Deregistration(request)['deregistrationResponse']
+
+    # Check the deregistration response
+    self.assertEqual(len(response), 2)
+    for x in range(0, 2):
+      self.assertEqual(response[x]['cbsdId'], cbsd_ids[x])
+      self.assertEqual(response[x]['response']['responseCode'], 0)
+    del request, response
+
+     # Prepare grant requests for two cbsdIds
+    grant_1 = json.load(open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_1['cbsdId'] = cbsd_ids[0]
+    grant_2 = json.load(open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_2['cbsdId'] = cbsd_ids[1]
+
+    # Send the grant requests
+    request = {'grantRequest': [grant_1, grant_2]}
+    response = self._sas.Grant(request)['grantResponse']
+
+    # Check the grant responses
+    # valid cbsdIds, responseCode should be 103 for both cbsdIds
+    self.assertEqual(len(response), 2)
+    for x, resp in enumerate(response):
+      self.assertEqual(resp['cbsdId'], cbsd_ids[x])
```

Comment:
I just noticed that the spec here doesn't seem to make much sense. If the CBSD has been deregistered, the CBSD ID is no longer valid and therefore should not be returned. Do you agree? (Your code matches the spec 100%, I'm just not sure the spec is right.)

Reply:
That's what happens when you follow directions!.....You are right , the CBSDs were deregistered so they should no longer exist in the SAS.  When the SAS responds to a non-existent CBSD ID request there should be no CBSD ID in the response (GRA3, p39)

Reply:
It's good that you did, otherwise I wouldn't have caught the error in the spec :) I will bring it up at today's WG4 meeting and one way or another hopefully we can get it changed. I think you have two options: (1) submit now and make a small PR later to remove this line, assuming I'm successful or (2) wait until this evening when I can give a status report from WG4 and you might be able to save yourself a small amount of work. Your choice :)

Reply:
I choose #2..thanks!!

Reply:
Good news: they approved this change (to remove the CBSD ID check) this afternoon!